src/app/config.rs

```diff
@@ -173,6 +185,13 @@ impl Config {
         }
 
         let config: Config = builder.extract()?;
+
+        if config.retry_vector_errors_forever && config.vector_max_retries.is_some() {
+            return Err(anyhow::anyhow!(
+                "vector_max_retries must be set to None when retry_vector_errors_forever is true"
+            ));
+        }
```

Comment:
I think we should warn here in sentry rather than failling. It'd be easy to have a deploy fail because someone forgot to set `vector_max_retries`, but it's not really hurting anything if it's set.

Reply:
i've removed the `vector_max_retries` option. just have the retry option now